src/main/java/subsystems/leds/Leds.java

```diff
@@ -0,0 +1,40 @@
+package subsystems.leds;
+
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj.AddressableLED;
+import frc.robot.Ports;
+
+
+public class Leds extends SubsystemBase {
+   public static Leds INSTANCE;
+   private final Color Yellow = new Color(254, 202, 82);
+   private final Color Purple = new Color(51, 1, 176);
+
+    public static Leds getInstance() {
+        if(INSTANCE == null){
+            INSTANCE = new Leds();
+        }
+        return INSTANCE;
+    }
+
+    public AddressableLED leds = new AddressableLED(Ports.Leds.LED);
+   public AddressableLEDBuffer ledBuffer = new AddressableLEDBuffer(Constants.LED_LENGTH);
+
+   public Leds(){
+       leds.setLength(ledBuffer.getLength());
+   }
+   public void setYellow(){
+        for(int i = 0; i<ledBuffer.getLength(); i++ ){
+            ledBuffer.setLED(i, Yellow);
+        }
+    }
+    public void setPurple(){
+        for(int i = 0; i<ledBuffer.getLength(); i++ ){
+            ledBuffer.setLED(i, Purple);
+
+        }
```

Comment:
The functions are still almost identical. Do you have an idea that will reduce repetition?
Remember, [Don't Repeat Yourself](https://en.wikipedia.org/wiki/Don't_repeat_yourself).